src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8635,21 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                int error = Marshal.GetLastWin32Error();
+
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
+                // If a directory has no alternate data streams, FindFirstStreamW returns ERROR_HANDLE_EOF.
+                if (error == NativeMethods.ERROR_HANDLE_EOF)
+                {
+                    return alternateStreams;
+                }
+
+                // Filesystems other than NTFS generally don't support alternate streams, so we still
+                // must be prepared to throw the error. It's usually ERROR_INVALID_PARAMETER in that case.
+                throw new Win32Exception(error);
```

Comment:
Is it possible to write user-friendly error message like "The file system does not support alternative streams."?

Reply:
Honestly, I'd prefer to decouple the underlying p/invokes into their own class, and write a more user-friendly middleware class that throws a message like that.

Also, Win32Exception is not the correct type of exception to be thrown if you want a more user-friendly error message. What type of exception do you want to be thrown on "File System Operation Not Supported"?  (this exception would also be useful for such things as trying to encrypt files on non-encryptable filesystems, trying to set files to be sparse on non-sparsable filesystems, trying to create symbolic or hard links on filesystems that don't support them, etc.)

Reply:
Usually we use a test hook. But it is edge case, not hot path, so I think we could skip the test.